plugins/turing/Makefile

```diff
@@ -0,0 +1,46 @@
+# ==================================
+# General
+# ==================================
+
+.PHONY: version
+version:
+	$(eval VERSION=$(if $(OVERWRITE_VERSION),$(OVERWRITE_VERSION),v$(shell ../../scripts/vertagen/vertagen.sh -f docker)))
+	@echo "version:" $(VERSION)
+
+.PHONY: vendor
+vendor:
+	@echo "Fetching dependencies..."
+	go mod vendor
+
+# ==================================
+# Code dependencies recipes
+# ==================================
+
+tidy-plugin:
+	go mod tidy
+
+# ==================================
+# Build recipes
+# ==================================
+
+.PHONY: build
+build:
+	@echo "Building binary..."
+	go build -o ./bin/plugin
+
+.PHONY: build-image
+build-image: vendor version
+	@$(eval IMAGE_TAG = $(if $(DOCKER_REGISTRY),$(DOCKER_REGISTRY)/,)xp-engine-plugin:${VERSION})
```

Comment:
`xp-engine-plugin` to be published to ghcr - https://github.com/gojek/turing-experiments/runs/7443640421?check_suite_focus=true (successful run)